infra/getting_started/your_first_lambda_function.md

```diff
@@ -0,0 +1,187 @@
+# [Getting Started](./README.md) / Deploying your first AWS Lambda Function with Go and Terraform
+
+## Intro
+This tutorial focuses on deploying a "Hello, World" Go program as a Lambda function using Terraform. The first section will cover creating the program in Go and the second section will cover writing the Terraform script that will deploy the program into AWS. As a bonus, we'll cover how you can configure AWS CloudWatch to invoke your Lambda function on a recurring schedule.
+
+## Assumptions
+* You're at least on macOS Mojave 10.14
+* You have the following installed on your machine:
+  * Terraform
+  * AWS CLI
+  * Go
+* You have an AWS account.
+* You have awscli configured with you credentials.
+* You know some Terraform basics like "plan" and "apply".
+
+## Creating the Go program
+1. First things first, open up that command line.
+2. Create a directory wherever you desire and name it `hello-lambda`.
+3. Set your current directory to `hello-lambda`.
+4. AWS has kindly provided a Go library to develop Lambda functions. Let's install it.
+
+    ```
+    go get github.com/aws/aws-lambda-go/lambda
+    ```
+
+5. Create a new file called `main.go`. This will contain the code that runs in our Lambda function.
+6. Within `main.go`:
+
+    ```
+    package main
+
+    import (
+        "github.com/aws/aws-lambda-go/lambda"
+    )
+
+    func hello() (string, error) {
+        return "Hello, world!", nil
+    }
+
+    func main() {
+        lambda.Start(hello)
+    }
+
+    ```
+
+    Let's dissect this program for a second.
+
+    As you can see, we imported the AWS library we installed. We then utilize it to call `lambda.Start()`. This takes a handler function, effectively calling it, and passes Lambda requests through it. 
+
+    The handler in this case is our function `hello`. This handler is relatively simple. It doesn't take any internal Lambda requests and just returns the classic "Hello, world!".
```

Comment:
What is an "internal lambda request"?

Reply:
The example we use should have a clearer method signature that accepts a context argument, like the Golang Lambda handler [example](https://docs.aws.amazon.com/lambda/latest/dg/go-programming-model-handler-types.html):

```go
// ...
func hello(ctx context.Context) (string, error) {
    // ...
}
```
Then we could improve the explanation like this:

> The handler in this case is our function `hello`. This handler is relatively simple, and doesn't use the Lambda context passed to it by `Lambda.start()` or make any other external requests. It only prints, "Hello, world!".

@ntwyman I have separate question: what's the golang convention for referring to function names in prose explanations? Above, I use `hello`, but I've also seen `hello()`, which probably makes it clearer that it's a function call and not another identifier. I'm coming from Ruby where we use `ClassName#instance_method` and `ClassName.class_method`.